net: l2: ethernet: ethernet support for AF_PACKET/SOCK_RAW/IPPROTO_RAW

jhirsi opened the pull request:
Ethernet support for AF_PACKET/SOCK_RAW/IPPROTO_RAW: setting protocol type for raw packets to be sent.

Signed-off-by: Jani Hirsimäki <jani.hirsimaki@nordicsemi.no>

jukkar reviewed (commented):
The change looks good, but the commit message could be enhanced a bit. Now it just states the obvious (what is being done and that can be seen in the patch itself), but fails to tell why the change is being done. Could you add more information to the commit message why this change is needed.
So it could be something like 
```
Setting protocol type for raw packets to be sent so that ....
```
or similar.

jhirsi committed:
net: l2: ethernet: ethernet support for AF_PACKET/SOCK_RAW/IPPROTO_RAW

Setting the protocol type for raw IP packets to be sent so that they
can be passed to Ethernet.

Signed-off-by: Jani Hirsimäki <jani.hirsimaki@nordicsemi.no>

jhirsi commented:
> The change looks good, but the commit message could be enhanced a bit. Now it just states the obvious (what is being done and that can be seen in the patch itself), but fails to tell why the change is being done. Could you add more information to the commit message why this change is needed. 

Thanks for the comments. OK, tried to elaborate the commit message bit and had to make the condition more strict to get existing 
socket_packet.test_raw_and_dgram_socket_exchange test passed (AF_PACKLET/SOCK_RAW/ETH_P_ALL).
